.github/scripts/changed-modules.sh

```
@@ -0,0 +1,84 @@
+unset IFS
+
+# merge all modules and examples into a single array
+allModules=("${modules[@]}")
+
+# sort allModules array
+IFS=$'\n' allModules=($(sort <<<"${allModules[*]}"))
+unset IFS
+
+# Get the list of modified files, retrieved from the environment variable ALL_CHANGED_FILES.
+# On CI, this value will come from a Github Action retrieving the list of modified files from the pull request.
+readonly modified_files=${ALL_CHANGED_FILES[@]}
```

Shellcheck (0.10.0)

[warning] Assigning an array to a string! Assign as array, or use * instead of @ to concatenate.

(SC2124)